### PR TITLE
Use assert pattern in test code of exporterhelper

### PR DESCRIPTION
**Pull request by huikang**
address #266

Signed-off-by: Hui Kang <kangh@us.ibm.com>

### CODE REVIEW

**Comment by huikang** [comment]
This is the first attempt to address #266 . Could anyone help to take a look? If it is on the right direction, I will complete the change for the exporter package. Thanks.

**Comment by ccaraman** [comment]
I will take first pass.

File: exporter/exporterhelper/tracehelper_test.go [hunk 1]
```diff
@@ -39,15 +40,13 @@ const (
 // TODO https://github.com/open-telemetry/opentelemetry-service/issues/266
 // Migrate tests to use testify/assert instead of t.Fatal pattern.
 func TestTraceExporter_InvalidName(t *testing.T) {
-	if _, err := NewTraceExporter("", newPushTraceData(0, nil)); err != errEmptyExporterName {
-		t.Fatalf("NewTraceExporter returns: Want %v Got %v", errEmptyExporterName, err)
-	}
+	_, err := NewTraceExporter("", newPushTraceData(0, nil))
```
Comment (ccaraman) [inline exporter/exporterhelper/tracehelper_test.go]: name a variable for the first return and assert it is nil.

File: exporter/exporterhelper/tracehelper_test.go [hunk 2]
```diff
@@ -39,15 +40,13 @@ const (
...
+	require.Equalf(t, errEmptyExporterName, err, "NewTraceExporter returns: Want %v Got %v", errEmptyExporterName, err)
```
Comment (ccaraman) [inline exporter/exporterhelper/tracehelper_test.go]: you can achieve the same with `assert.Equal(t, errEmptyExporterName, err)` The print out of the error shows expected value and actual when this isn't true so no need to add the extra text.

File: exporter/exporterhelper/tracehelper_test.go [hunk 3]
```diff
@@ -65,61 +64,53 @@ func TestTraceExporter_Default_ReturnError(t *testing.T) {
...
 	checkRecordedMetricsForTraceExporter(t, te, nil, 1)
 }
 
 func TestTraceExporter_WithRecordMetrics_ReturnError(t *testing.T) {
 	want := errors.New("my_error")
 	te, err := NewTraceExporter(fakeExporterName, newPushTraceData(0, want), WithRecordMetrics(true))
-	if err != nil {
-		t.Fatalf("NewTraceExporter returns: Want nil Got %v", err)
-	}
+	require.Nilf(t, err, "NewTraceExporter returns: Want nil Got %v", err)
```
Comment (ccaraman) [inline exporter/exporterhelper/tracehelper_test.go]: Add a check for te not being nil `require.NotNil(t, te)`

File: exporter/exporterhelper/tracehelper_test.go [hunk 4]
```diff
@@ -161,27 +152,24 @@ func checkRecordedMetricsForTraceExporter(t *testing.T, te exporter.TraceExporte
 	ctx := observability.ContextWithReceiverName(context.Background(), fakeReceiverName)
 	const numBatches = 7
 	for i := 0; i < numBatches; i++ {
-		if err := te.ConsumeTraceData(ctx, td); err != wantError {
-			t.Fatalf("Want %v Got %v", wantError, err)
-		}
+		err := te.ConsumeTraceData(ctx, td)
```
Comment (ccaraman) [inline exporter/exporterhelper/tracehelper_test.go]: nit: you can make this one line and have the method invoked in the require.Equal

**Comment by ccaraman** [comment]
This is looking good thank you! Please apply the comments to all applicable places where you are making these changes.

File: exporter/exporterhelper/common_test.go [hunk 5]
```diff
@@ -36,26 +38,17 @@ func TestWithSpanName(t *testing.T) {
...
-	if g, w := errToStatus(nil), okStatus; g != w {
-		t.Fatalf("Status: Want %v Got %v", w, g)
-	}
-
-	errorStatus := trace.Status{Code: trace.StatusCodeUnknown, Message: "my_error"}
-	if g, w := errToStatus(errors.New("my_error")), errorStatus; g != w {
-		t.Fatalf("Status: Want %v Got %v", w, g)
-	}
+	errorStatus := errToStatus(nil)
+	require.Equalf(t, okStatus, errorStatus, "Status: Want %v Got %v", okStatus, errorStatus)
```
Comment (pjanotti) [inline exporter/exporterhelper/common_test.go]: Use `require.Equal` and do not require the message, in general avoid the `f` functions. Apply this to the other locations, I think that the message is just worth if is something that requires some explanation.

File: exporter/exporterhelper/tracehelper_test.go (hunk 1 above)
Comment (pjanotti) [reply exporter/exporterhelper/tracehelper_test.go]: besides the refactor this becomes an improvement in the test, the same apply to other locations below.

File: exporter/exporterhelper/tracehelper_test.go (hunk 2 above)
Comment (pjanotti) [reply exporter/exporterhelper/tracehelper_test.go]: Just mentioned the same above...

**Pull request merged**